File: app/src/main/java/com/example/notify/authenticationpage.java
package com.example.notify;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.app.Activity;
import android.graphics.Color;
import android.graphics.Typeface;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.google.firebase.auth.FirebaseAuth;

public class authenticationpage extends AppCompatActivity {
    private String email,password;
    private TextView login_selector,sign_selector,forget_selector;
    private Button login_button,sign_button;
    LinearLayout login_layout,sign_layout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_authenticationpage);

        FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();
        EditText user_email_address = findViewById(R.id.user_email_address);
        EditText user_password = findViewById(R.id.user_password);

        login_selector = findViewById(R.id.login_selector);
        sign_selector = findViewById(R.id.sign_selector);
        login_layout = findViewById(R.id.login_linear_layout);
        sign_layout = findViewById(R.id.sign_linear_layout);
        login_layout.setVisibility(View.GONE);


        loginFragmentManager();
        login_selector.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loginFragmentManager();
                login_layout.setVisibility(View.GONE);
                sign_layout.setVisibility(View.VISIBLE);
            }
        });

        sign_selector.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                signFragmentManager();
                sign_layout.setVisibility(View.GONE);
                login_layout.setVisibility(View.VISIBLE);
            }
        });




    }

    void loginFragmentManager(){
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.setCustomAnimations(android.R.anim.slide_in_left,android.R.anim.slide_out_right);
        fragmentTransaction.replace(R.id.authframe,new LoginFragment(),"h");
        fragmentTransaction.commit();
        login_selector.setTypeface(null,Typeface.BOLD);
        sign_selector.setTypeface(null,Typeface.NORMAL);
    }
    void signFragmentManager(){
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.setCustomAnimations(R.anim.enter,R.anim.exit);
        fragmentTransaction.replace(R.id.authframe,new SignFragment(),"h");
        fragmentTransaction.commit();
        sign_selector.setTypeface(null,Typeface.BOLD);
        login_selector.setTypeface(null,Typeface.NORMAL);
    }
    void forgetFragmentManager(){
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.authframe,new forgetFragment());
        fragmentTransaction.commit();
    }
}

File: app/src/main/java/com/example/notify/Authenticationpage.kt
package com.example.notify

import android.graphics.Typeface
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.notify.databinding.ActivityAuthenticationpageBinding
import com.google.firebase.auth.FirebaseAuth

class authenticationpage : AppCompatActivity() {
    private lateinit var binding: ActivityAuthenticationpageBinding
    private val email: String? = null
    private val password: String? = null
    var login_layout: LinearLayout? = null
    var sign_layout: LinearLayout? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_authenticationpage)
        val firebaseAuth: FirebaseAuth = FirebaseAuth.getInstance()
        val user_email_address = findViewById<EditText>(R.id.user_email_address)
        val user_password = findViewById<EditText>(R.id.user_password)

        binding.loginLinearLayout.setVisibility(View.GONE)
        loginFragmentManager()
        binding.loginSelector.setOnClickListener(View.OnClickListener {
            loginFragmentManager()
            binding.loginLinearLayout.visibility = View.GONE
            binding.signLinearLayout.visibility = View.VISIBLE
        })
        binding.signSelector.setOnClickListener(View.OnClickListener {
            signFragmentManager()
            binding.signLinearLayout.visibility = View.GONE
            binding.loginLinearLayout.visibility = View.VISIBLE
        })
    }

    fun loginFragmentManager() {
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.setCustomAnimations(
            android.R.anim.slide_in_left,
            android.R.anim.slide_out_right
        )
        fragmentTransaction.replace(R.id.authframe, LoginFragment(), "h")
        fragmentTransaction.commit()
        binding.loginSelector.setTypeface(null, Typeface.BOLD)
        binding.signSelector.setTypeface(null, Typeface.NORMAL)
    }

    fun signFragmentManager() {
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.setCustomAnimations(R.anim.enter, R.anim.exit)
        fragmentTransaction.replace(R.id.authframe, SignFragment(), "h")
        fragmentTransaction.commit()
        binding.signSelector.setTypeface(null, Typeface.BOLD)
        binding.loginSelector.setTypeface(null, Typeface.NORMAL)
    }

    fun forgetFragmentManager() {
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.authframe, forgetFragment())
        fragmentTransaction.commit()
    }
}